### PR TITLE
fix(savings-goals): cap progress label at "Funded" for overfunded goals

### LINKED ISSUE #112: Handle overfunded savings goals in SavingsGoalListView
In `SavingsGoalListView`, when `fundedAmount` exceeds `targetAmount`, `progressPercent` will compute a value above 100. The `Bar` component clamps its visual bar to 100%, but the text label next to it will display the raw percentage (e.g. `110%`), creating a visual inconsistency where the bar shows full and the label shows more than full.

**Acceptance criteria**
- When `fundedAmount >= targetAmount`, display the progress label as `100%` (or a "Funded" badge) and the bar as fully filled
- The fix should be covered by a test asserting the display for an overfunded goal

Deferred from [#107](https://github.com/rmartz/personal-budget/pull/107) during code review.

---
*Created by Claude Sonnet 4.6*

### PR DESCRIPTION
**Pull request by rmartz**
When a savings goal's `fundedAmount` exceeds its `targetAmount`, the progress bar (`Bar` component) already clamps the visual fill to 100%, but the text label next to it displayed the raw calculated percentage (e.g. "110%"), creating a visual inconsistency.

`SavingsGoalListItemView` now checks `fundedAmount >= targetAmount` and renders a `"Funded"` label instead of the raw percentage for any fully or over-funded goal. The `Bar` value is also explicitly clamped with `Math.min(progressPercent, 100)` (making the intent explicit, even though `Bar` already clamps internally). A `fundedLabel` constant was added to `SAVINGS_GOAL_LIST_ITEM_COPY`. A `Funded` and `Overfunded` Storybook story were added to `SavingsGoalListItem.stories.tsx`.

## Tests
3 tests added (funded-equals-target, overfunded, no-raw->100-label), all passing. Full suite: 869/869 passing.

Closes #112

---
*Created by Claude Sonnet 4.6*

### CODE REVIEW

**Review by rmartz — COMMENTED** [review]
## Requested changes

- **`isFunded` missing `targetAmount > 0` guard** — The new `isFunded` constant on line 71 is defined as `goal.fundedAmount >= goal.targetAmount`, but the pre-existing `progressPercent` calculation already guards against `targetAmount === 0` (returning 0 in that case). When `targetAmount === 0` and `fundedAmount === 0`, `isFunded` evaluates to `true` (since `0 >= 0`), which would display "Funded" while the bar shows 0% — a visual inconsistency. Both the create and edit dialogs currently validate `targetAmount > 0`, so this edge case cannot arise through normal UI flows today, but the type allows it and a future service-layer path or test fixture could produce it. The fix is a one-liner: `const isFunded = goal.targetAmount > 0 && goal.fundedAmount >= goal.targetAmount`. This aligns the guard with the existing `progressPercent` pattern and closes the inconsistency. This is also the concern raised in the open Copilot thread at [SavingsGoalListItem.tsx line 71](https://github.com/rmartz/personal-budget/pull/179#discussion_r2091283729).

## Verdict

One required change: align `isFunded` with the existing `targetAmount > 0` guard used by `progressPercent`.

---
*Claude Sonnet 4.6*

**Review by rmartz — COMMENTED** [review]
## Prior items

- ✅ `isFunded` missing `targetAmount > 0` guard — fixed in commit `d5c0c83` with `const isFunded = goal.targetAmount > 0 && goal.fundedAmount >= goal.targetAmount`, and a test for the zero-target case was added.

## Requested changes

- **`progressPercent` uses `Math.round`, producing `100` before the goal is funded** — [`SavingsGoalListItem.tsx` line 71](https://github.com/rmartz/personal-budget/pull/179#discussion_r3254961932). `Math.round(4999 / 5000 * 100)` produces `100`, but since `isFunded` is false (4999 < 5000), `progressLabel` renders `"100%"` rather than `"Funded"` — the bar shows full but the label reads as a percentage, which is inconsistent and misleading. `GoalCard` already handles this correctly: when not funded, it uses `Math.min(99, Math.floor(...))` to guarantee the displayed percent stays below 100 until the goal is actually funded. Apply the same pattern here so `progressPercent` can only equal 100 when `isFunded` is true.

- **`SavingsGoalList.spec.tsx` line 183 assertion gives false confidence** — [`SavingsGoalList.spec.tsx` line 183](https://github.com/rmartz/personal-budget/pull/179#discussion_r3254961950). `screen.getAllByText(SAVINGS_GOAL_LIST_ITEM_COPY.fundedLabel)` matches every element containing `"Funded"`, including the column header (`SAVINGS_GOAL_LIST_COPY.columnFunded` is also `"Funded"`). Calling `.toBeDefined()` on the returned array always passes, even if only the header matched and the row-level label is absent. Fix this by either (a) asserting `screen.getAllByText(SAVINGS_GOAL_LIST_ITEM_COPY.fundedLabel).length` is greater than `1`, confirming the label appears beyond the header, or (b) scoping with `within(goalRow).getByText(...)` to verify the label is present specifically in the goal row.

## Verdict

Two issues remain: a correctness bug where `Math.round` lets `progressPercent` reach `100` for goals that are not yet funded, and a test assertion that can pass without actually verifying the row-level "Funded" label.

---
*Claude Sonnet 4.6*

File: src/components/savings-goals/SavingsGoalList.spec.tsx [hunk 1]
```diff
@@ -177,7 +177,11 @@ describe("SavingsGoalListView", () => {
           onReorder={noop}
         />,
       );
-      expect(screen.getByText("100%")).toBeDefined();
+      // Column header also reads "Funded"; assert length > 1 to confirm the
+      // label appears in at least one goal row in addition to the header.
+      expect(
+        screen.getAllByText(SAVINGS_GOAL_LIST_ITEM_COPY.fundedLabel).length,
+      ).toBeGreaterThan(1);
```
[the remark this answers is not in the document]
Comment (rmartz) [reply src/components/savings-goals/SavingsGoalList.spec.tsx:184]: The brittleness here is a false-negative risk (the test would start failing on a copy drift that doesn't change behavior), not a false-positive risk (the test silently passing while the row label is absent). A `.length > 1` failure is loud and immediately diagnosable. The prior review explicitly offered both option (a) `.length > 1` and option (b) `within` scoping as valid alternatives; fix-review applied (a). Since both approaches adequately guard against the original correctness bug (the assertion that always passed on the header alone), and the risk Copilot describes would produce a noisy test failure rather than a silent false pass, I'm treating this as a style preference rather than a required change.

— *Claude Sonnet 4.6*

**Review by rmartz — COMMENTED** [review]
## Prior items

- ✅ `progressPercent` uses `Math.round`, producing `100` before the goal is funded — fixed in the fix-review commit: `isFunded` is now computed first, then `progressPercent = isFunded ? 100 : goal.targetAmount > 0 ? Math.min(99, Math.floor(...)) : 0`, exactly matching the `GoalCard` pattern.
- ✅ [`SavingsGoalList.spec.tsx` line 183 assertion gives false confidence](https://github.com/rmartz/personal-budget/pull/179#discussion_r3254961950) — fixed: `.toBeDefined()` on the array replaced with `.length` `.toBeGreaterThan(1)`, confirming the label appears beyond the header alone.
- ✅ [New Copilot thread — `SavingsGoalList.spec.tsx` line 184](https://github.com/rmartz/personal-budget/pull/179#discussion_r3255028044): assertion brittleness if copy strings diverge — dismissed. The risk is a false negative (a loud test failure), not a false positive (silent pass while the row label is absent). The prior review explicitly offered both `.length > 1` (option a) and `within` scoping (option b) as valid alternatives; fix-review applied option (a). Replied and resolved.

## Verdict

All prior items resolved, new Copilot thread dismissed with reasoning — ready to merge after UAT.

---
*Claude Sonnet 4.6*

**Comment by rmartz** [comment]
**Pre-merge check**

- ✅ No WIP or draft
- ✅ PR approved, no changes requested, no open threads
- ✅ UAT complete (skip-uat)
- ✅ Branch up to date, no conflicts
- ✅ All CI green

Merging.

---
*Posted by Claude Sonnet 4.6*
